### PR TITLE
Pytest & Nox

### PR DESCRIPTION
**Pull request by javdrher**
As it seems that the nose project is no longer active and I encountered a lot of problems with it in another project, I switched the testing framework to py.test. Additionally, I added support for nox: it makes testing very easy as it takes care of creation of virtualenvs and all the dependencies.

### CODE REVIEW

**Comment by javdrher** [comment]
Could you review this PR @icouckuy ?

**Comment by icouckuy** [comment]
Yes, sorry. It has been on my mind but the internet is very flaky here. Is it blocking other PRs?

File: nox.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        '--cov-config=.coveragerc',
+        'testing/unit'
+    )
+
+@nox.session
+def system(session):
+    session.install('pytest', 'pytest-cov', *TEST_DEPS)
+    session.install('-e', '.', '--process-dependency-links')
+
+    # Run py.test against the unit tests.
```
Comment (icouckuy) [inline nox.py]: against the system tests

File: nox.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import nox
+
+TEST_DEPS = ('six', 'parameterized', 'nbconvert', 'nbformat', 'jupyter', 'jupyter_client', 'matplotlib', 'nose')
```
Comment (icouckuy) [inline nox.py]: k, as discussed we could split matplotlib from this as it is only needed for the system tests

**Pull request merged**